### PR TITLE
Geometry info crop xyz20230413

### PR DESCRIPTION
**Pull request by LLipter**
Add `CropContainerMarginXYZ` field to `GeometryInfo` object, which is a list of 6 floating numbers.

### CODE REVIEW

File: include/openrave/kinbody.h [hunk 1]
```diff
@@ -359,7 +359,13 @@ class OPENRAVE_API KinBody : public InterfaceBase
                    && _fTransparency == other._fTransparency
                    && _bVisible == other._bVisible
                    && _bModifiable == other._bModifiable
-                   && _calibrationBoardParameters == other._calibrationBoardParameters;
+                   && _calibrationBoardParameters == other._calibrationBoardParameters
+                   && _vCropContainerMarginXYZ[0] == other._vCropContainerMarginXYZ[0]
```
Comment (ziyan) [inline include/openrave/kinbody.h:363]: boost array has no `operator==` overload?

File: include/openrave/kinbody.h [hunk 2]
```diff
@@ -490,6 +496,7 @@ class OPENRAVE_API KinBody : public InterfaceBase
         float _fTransparency = 0; ///< value from 0-1 for the transparency of the rendered object, 0 is opaque
         bool _bVisible = true; ///< if true, geometry is visible as part of the 3d model (default is true)
         bool _bModifiable = true; ///< if true, object geometry can be dynamically modified (default is true)
+        boost::array<dReal, 6> _vCropContainerMarginXYZ = {0, 0, 0, 0, 0, 0};
```
Comment (ziyan) [inline include/openrave/kinbody.h:499]: It's `XYZXYZ`, not `XYZ`

**Comment by undisputed-seraphim** [comment]
Can/Should we put the operators in `geometry.h`?

Alternatively, should we also write fuller 6-dim vector implementations so that it looks more consistent with `RaveVector` in `geometry.h`? (Actually I think we should use `Eigen3` vectors and matrices everywhere so we don't have to keep reinventing the wheel each time we need a new N-dim vector)

File: include/openrave/kinbody.h [hunk 3]
```diff
@@ -490,6 +517,7 @@ class OPENRAVE_API KinBody : public InterfaceBase
         float _fTransparency = 0; ///< value from 0-1 for the transparency of the rendered object, 0 is opaque
         bool _bVisible = true; ///< if true, geometry is visible as part of the 3d model (default is true)
         bool _bModifiable = true; ///< if true, object geometry can be dynamically modified (default is true)
+        boost::array<dReal, 6> _vCropContainerMarginsXYZXYZ = {0, 0, 0, 0, 0, 0};
```
Comment (rdiankov) [inline include/openrave/kinbody.h:520]: I think the 6 dim array has been hard to understand in the system and will be even harder if introduced into openrave this way. How about `Vector negCropContainerMargins, posCropContainerMargins`?

File: src/libopenrave/kinbodygeometry.cpp [hunk 4]
```diff
@@ -1031,6 +1053,22 @@ void KinBody::GeometryInfo::DeserializeJSON(const rapidjson::Value &value, const
                 _meshcollision.Clear();
             }
         }
+        if (value.HasMember("negativeCropContainerMargins")) {
+            orjson::LoadJsonValueByKey(value, "negativeCropContainerMargins", vGeomDataTemp);
+            vGeomDataTemp *= fUnitScale;
+            if (vGeomDataTemp != _vNegativeCropContainerMargins) {
+                _vNegativeCropContainerMargins = vGeomDataTemp;
+                _meshcollision.Clear();
```
Comment (ziyan) [inline src/libopenrave/kinbodygeometry.cpp:1061]: You do not need to clear mesh here since this parameter does not affect mesh generation.

Comment (LLipter) [reply src/libopenrave/kinbodygeometry.cpp]: ok, they are removed.

**Comment by rdiankov** [comment]
for now accepting, but would like visualizations.

**Pull request merged**